### ISSUE TITLE
Allow for custom copyright messages in WNDB export

### ISSUE THREAD

**Issue by jmccrae** [jmccrae/gwn-scala-api#14]
Some projects would like to release in WNDB, but the copyright message is hard-coded to be the Princeton message. We should add a parameter to allow custom messages. These messages must be 1740 characters or cut or padded to this length,

**Comment by ekaf** [comment]
Yes, traditionally the first synset in each pos.data file used to start at byte offset 1740, because that was the length of the Princeton copyright notice. But does this number really play a role today? Synset offsets are arbitrary, so why couldn't the copyright text be any other length?

**Comment by jmccrae** [comment]
Perhaps a flag to allow both options would be the best implementation? I fear some parsers may just drop the first 1740 bytes out of the file without checking

**Comment by ekaf** [comment]
Hard-coding numbers is generally dangerous, and I am not aware of anybody who does it. NLTK uses the following:
>                    if line.startswith(" "):
>                        continue
It is true that people are accustomed to finding a synset at offset 1740, but things should be allowed to evolve.

**Comment by 1313ou** [comment]
Specs say:
"""
Each data/index file begins with several lines containing a copyright
notice, version number, and license agreement. These lines all begin with
two spaces and the line number.
"""
https://wordnet.princeton.edu/documentation/wndb5wn

All parsers have to abide by specifications. I don't know of any that takes
the 1740 offset for granted.

**Comment by ekaf** [comment]
Yes, the 1740 limit is not likely to be a problem for parsers. However, it is very convenient to be able to take the existence of the 1740 synset offset for granted, in order to provide an example in documentation. Consider this documentation string in the NLTK wordnet module:

>    def synset_from_pos_and_offset(self, pos, offset):
        """
[...]
        >>> from nltk.corpus import wordnet as wn
        >>> print(wn.synset_from_pos_and_offset('n', 1740))
        Synset('entity.n.01')
        """

If no synset is available at offset 1740, this test will fail. And if no particular offset is certain across wordnet versions, no example could be provided in documentation. This may not be a great loss, though in my opinion, it is a reason why it is still desirable to keep the 1740 offset, if possible
